src/PriorityQueue.js

```diff
@@ -0,0 +1,38 @@
+class PriorityQueue {
+  constructor() {
+    this.data = []
+    this.head = 0
+  }
+
+  enqueue(element) {
+    this.data[this.head] = element
+    this.head++
+  }
+
+  dequeue() {
+    return this.head > 0 ? this.data.shift() : null
+  }
+
+  front() {
+    return this.head > 0 ? this.data[0] : null
+  }
+
+   back() {
+    return this.head > 0 ? this.data[this.head-1] : null
+   }
+
+  isEmpty() {
+    return this.head === 0
+  }
+
+  length() {
+    return this.head > 0 ? this.head : null
```

Comment:
A `PriorityQueue` assigns a priority to some data and enqueues the data based on that priority value. I think there's a misunderstanding here.